src/Kiota.Builder/Refiners/TypeScriptRefiner.cs

```diff
@@ -10,73 +10,68 @@
 namespace Kiota.Builder.Refiners;
 public class TypeScriptRefiner : CommonLanguageRefiner, ILanguageRefiner
 {
-    public TypeScriptRefiner(GenerationConfiguration configuration) : base(configuration) {}
+    public TypeScriptRefiner(GenerationConfiguration configuration) : base(configuration) { }
     public override Task Refine(CodeNamespace generatedCode, CancellationToken cancellationToken)
     {
-        return Task.Run(() => {
+        return Task.Run(() =>
+        {
             cancellationToken.ThrowIfCancellationRequested();
             ReplaceIndexersByMethodsWithParameter(generatedCode, false, "ById");
             RemoveCancellationParameter(generatedCode);
             CorrectCoreType(generatedCode, CorrectMethodType, CorrectPropertyType, CorrectImplements);
             CorrectCoreTypesForBackingStore(generatedCode, "BackingStoreFactorySingleton.instance.createBackingStore()");
             cancellationToken.ThrowIfCancellationRequested();
-            AddInnerClasses(generatedCode, 
-                true, 
+            AddInnerClasses(generatedCode,
+                true,
                 string.Empty,
                 true);
             // `AddInnerClasses` will have inner classes moved to their own files, so  we add the imports after so that the files don't miss anything.
             // This is because imports are added at the file level so nested classes would potentially use the higher level imports.
             AddDefaultImports(generatedCode, defaultUsingEvaluators);
-            DisableActionOf(generatedCode, 
+            DisableActionOf(generatedCode,
                 CodeParameterKind.RequestConfiguration);
             cancellationToken.ThrowIfCancellationRequested();
             AddPropertiesAndMethodTypesImports(generatedCode, true, true, true);
             AddParsableImplementsForModelClasses(generatedCode, "Parsable");
             ReplaceBinaryByNativeType(generatedCode, "ArrayBuffer", null, isNullable: true);
             cancellationToken.ThrowIfCancellationRequested();
             ReplaceReservedNames(generatedCode, new TypeScriptReservedNamesProvider(), x => $"{x}_escaped");
```

Comment:
@baywet 
I see this error -> https://github.com/microsoft/kiota-typescript/actions/runs/4024090844/jobs/6915715774#step:14:18

In case the model name is a reserved word, such as `package` or `case`, then "sometimes" the `ReplaceReservedNames` function doesn't sync well with the `Interface conversion` process. As a result, `case_escaped`  and/or `package_escaped` remain as classes instead of converting to interface. This error is not consistent. There are times when I am able to refine and generate a complete set of model interfaces with `case_escaped`  and `package_escaped` as interfaces. 
 
 Do you have any suggestion to solve this issue?

Reply:
I'm not able to find `packages/test/generatedCode/models/package_escaped.ts` on the `nikithauc/model-interface-serialization` branch at the moment, did you remove it?
Could you provide the generated code for one of those failures?

Reply:
I did not delete this any artifact.
Pointing out to the inconsistency, if you see this run was successful https://github.com/microsoft/kiota-typescript/actions/runs/4033992745/jobs/6934865542. 
Let me see if I can generate this. There is no consistent way of reproducing this issue.

Reply:
so are you referring to a race condition? I'm struggling to understand what the issue is at this point.

Reply:
I believe it is a race condition. 

I was able to get an error sample reproduces. 
Check the following:

1. Case_escaped converts  fine to an interface: 
https://github.com/microsoft/kiota-typescript/blob/package-esc-missing/packages/test/generatedCode/models/security/case_escaped.ts

2. Package_escaped doesn't convert to an interface as expected. 
https://github.com/microsoft/kiota-typescript/blob/package-esc-missing/packages/test/generatedCode/models/package_escaped.ts

I believe the issue is  `ReplaceReservedNames` doesn't complete before `IntroducesInterfacesAndFunctions` synchronously. Sometimes both these classes fail to convert while sometimes they both convert fine.

Reply:
Refiners are single threaded and continuous to avoid race conditions at that stage. The only places where multiple threads are in use are when we're building the code dom in kiota builder (before the refiner's execution) and after when we write the files.
Ordering of the refiners can sometimes impact the result, it might be worth exploring that avenue.

Reply:
Resolved the issue. While replacing the reserved names, the `InnerChild` Elements of the Parent block were not updated with the new name. Added a fix for that in this  PR.